### DISCUSSION TITLE
NanoMQ Message Size Length Limit?

### DISCUSSION THREAD

**Discussion by rokashevich** [nanomq/nanomq#528] [General]
Can not send more than 1.5Mb like this:
`nanomq pub  start -t foo -f file`
Forgive me, I'm new at this :)

**Comment by JaylinYu** [comment]
Hi, what broker are you publishing to? Please note that most of MQTT brokers has a Max message size limitation.

**Reply by rokashevich** [reply]
Hi @JaylinYu!
Thank you very much for your reply :)
I'm just using NanoMQ builtin broker: `nanomq broker start`
It's `property_size` -s command line option is not this Max message size limitation isn't it?
Btw, what's the maximum number of topics I can publish?
I'm just trying to figure out the non-hardware limitations of a such a promising broker like nanomq...
Thankx

**Reply by JaylinYu** [reply]
Hi @rokashevich , thank you for your interest in NanoMQ. Feel free to ask any IoT question here, no matter it is related to nanomq or not.
 For NanoMQ, the default max packet size is 2 * 1024 * 1024 bytes (in nng/include/nng/protocol/mqtt/mqtt.h),  I would like to make this public configurable soon.

**Reply by JaylinYu** [reply]
Ahh, what I forget to mention is, that there is also a max receive packet size limit from the client-side. it is defined in MQTT V5 as "Maximum Packet Size". 
For nanomq, if the client does not set its own Maximum Packet Size, then the broker will use a default value: "65535", this also affects when you do massive packet transferring. the default value is in "conn_param_init()".

After modify the max limit, I can pub/sub up to 256mb packets which is limit by MQTT protocol(4 bytes of length). Please tell me your detailed use case, so that I can see if there is any way to expose these configurations to help you.

**Comment by JaylinYu** [comment]
Hi @rokashevich  I have made these configurations exposed. Please have a try on 
nanomq.conf:
client_max_packet_size
max_packet_size

**Reply by rokashevich** [reply]
Thank you @JaylinYu!
